tests/ValidatorDelegates/CSPBadFlagsTest.php

```diff
@@ -0,0 +1,182 @@
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "default-src 'unsafe-inline' 'unsafe-eval'; "
+                . "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
```

Comment:
> Readability is king is my usual litmus test. Maybe subtype the errors in order to validate that the right list of types is generated, but then test the string output of each subtype in its own test? That's probably overkill, too.

It's certainly sometimes difficult to balance the simplicity with DRY.

IMO DRY works quite well for the actual code base, for tests, the most important thing is to be able to see easily see what's going on (preferably within one file, or method even).

I'll have a ponder about best way to balance this, kinda thinking it may be easiest, and most readable, to just reimplement [this string substitution](https://github.com/aidantwoods/SecureHeaders/blob/2.0/src/ValidatorDelegates/CSPBadFlags.php#L50-L54) as a method in the test file.

Maybe I'll push doing that to after release though, more of an aesthetics thing, so won't change any actual test results.

> Good luck with the release.

Thank you! 🎉